src/storage-client/src/controller.rs

```diff
@@ -2295,7 +2295,9 @@ where
                     .await
                     .expect("stash operation must succeed");
 
-                self.finalize_shards().await;
+                if self.state.config.finalize_shards {
+                    self.finalize_shards().await;
+                }
```

Comment:
Thoughts on logging some informational messages here?

```suggestion
                if self.state.config.finalize_shards {
                    info!("triggering shard finalization due to dropped storage object");
                    self.finalize_shards().await;
                } else {
                    info!("not triggering shard finalization due to dropped storage object because enable_storage_shard_finalization parameter is false")
                }
```

Reply:
related thought: a metric gauge for the # of awaiting-finalization shards would be handy